.github/workflows/ci.yaml

```diff
@@ -1,13 +1,11 @@
-name: Build OpenSearch from source and run tests
+name: Build and Test
 
 env:
   VERSION: 2.4.0
-  RELEASE: edge
 
 on:
-  push:
-    branches:
-      - main
+  workflow_call:
```

Comment:
Docs weren't clear, what's the gain for this?

Reply:
This is to allow `ci.yaml` to be called from another action, such as `release.yaml`, so that the logic is not duplicated among files, i.e: [here](https://github.com/canonical/opensearch-snap/blob/ci-fix/.github/workflows/release.yaml#L14-L16)